docs/guides/magicgui.md

```diff
@@ -552,3 +552,31 @@ class MyGui(FunctionGui):
 def napari_experimental_provide_dock_widget():
     return MyGui
 ```
+
+### `magicgui.widgets.create_widget`
+
+If you want to create a layer selection as [shown above](magicgui-and-type-annotations) but you cannot use the {func}`@magicgui <magicgui.magicgui>` decorator, you can create the selection widget with the {func}`create_widget <magicgui.widgets.create_widget>` function. In the following example we subclass `QtWidgets.QWidget` and add a widget that is annotated as {attr}`napari.types.ImageData`
```

Comment:
```suggestion
If you want to create a layer selection as [shown above](#magicgui-and-type-annotations) but you cannot use the {func}`@magicgui <magicgui.magicgui>` decorator, you can create the selection widget with the {func}`create_widget <magicgui.widgets.create_widget>` function. In the following example we subclass `QtWidgets.QWidget` and add a widget that is annotated as {attr}`napari.types.ImageData`:

```

Reply:
Based on my experience now I prefer to modify the example and suggest using `magic_factory` instead of magicgui.

Reply:
@Czaki can you elaborate here? The example doesn't use the magicgui decorator at all, which is what magic_factory replaces. So what would you suggest here?

Anyway TIL about create_widget so I like this! 😅

Reply:
I mean update this docstring to point magic factory.